### PR TITLE
feat: New Module `avm/res/communication/communication-service`

### PR DESCRIPTION
**Pull request by krbar**
## Description

New Module `avm/res/communication/communication-service`.

Related Module Proposal: https://github.com/Azure/Azure-Verified-Modules/issues/748

## Adding a new module

- [x] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Pipeline references

| Pipeline |
| - |
| [avm.res.communication.communication-service](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.communication.communication-service.yml) |

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -21,6 +21,7 @@
 /avm/res/cache/redis/ @Azure/avm-res-cache-redis-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/cdn/profile/ @Azure/avm-res-cdn-profile-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/cognitive-services/account/ @Azure/avm-res-cognitiveservices-account-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/communication/communication-service/ @Azure/avm-res-communication-communicationservice-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (eriqua) [inline .github/CODEOWNERS:24]: Placeholder. Please revolve after onboarding to MCR is complete

Comment (krbar) [reply .github/CODEOWNERS:24]: I just got an info from @matebarabas that the MCR onboarding is completed.

File: avm/res/communication/communication-service/tests/e2e/defaults/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      name: '${namePrefix}${serviceShort}001'
+      location: 'global'
```
Comment (AlexanderSehr) [inline avm/res/communication/communication-service/tests/e2e/defaults/main.test.bicep:45]: Should global be the default in the module?

**Review by AlexanderSehr — DISMISSED** [review]

**Comment by AlexanderSehr** [comment]
Hey @krbar, the PR looks fine to me, but seems to have conflicts (presumably because of the other communication module being merged first. Please resolve. If you need help with that, let me know.

**Comment by krbar** [comment]
> Hey @krbar, the PR looks fine to me, but seems to have conflicts (presumably because of the other communication module being merged first. Please resolve. If you need help with that, let me know.

Thank you @AlexanderSehr, the conflicts were expected and are already resolved. A final pipeline run is in progress.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**